servers/src/main/java/tachyon/worker/block/evictor/PartialLRUEvictor.java

```diff
@@ -0,0 +1,241 @@
+
+  /**
+   * access-ordered {@link java.util.LinkedHashMap} from blockId to {@link #UNUSED_MAP_VALUE}(just a
+   * placeholder to occupy the value), acts as a LRU double linked list where most recently accessed
+   * element is put at the tail while least recently accessed element is put at the head.
+   */
+  private Map<Long, Boolean> mPartialLRUCache = Collections
+      .synchronizedMap(new LinkedHashMap<Long, Boolean>(LINKED_HASH_MAP_INIT_CAPACITY,
+          LINKED_HASH_MAP_INIT_LOAD_FACTOR, LINKED_HASH_MAP_ACCESS_ORDERED));
+
+  public PartialLRUEvictor(BlockMetadataManager meta) {
+    mMeta = Preconditions.checkNotNull(meta);
+
+    // preload existing blocks loaded by StorageDir to Evictor
+    for (StorageTier tier : mMeta.getTiers()) {
+      for (StorageDir dir : tier.getStorageDirs()) {
+        for (long blockId : dir.getBlockIds()) {
+          mPartialLRUCache.put(blockId, UNUSED_MAP_VALUE);
+        }
+      }
+    }
+  }
+
+  @Override
+  public EvictionPlan freeSpace(long availableBytes, BlockStoreLocation location)
+      throws IOException {
+    List<Pair<Long, BlockStoreLocation>> toMove = new ArrayList<Pair<Long, BlockStoreLocation>>();
+    List<Long> toEvict = new ArrayList<Long>();
+
+    Pair<StorageDir, List<Long>> dirCandidate = getDirCandidate(availableBytes, location);
+
+    if (dirCandidate == null) {
+      LOG.error("Failed to freeSpace: No StorageDir has enough capacity of {} bytes",
+          availableBytes);
+      return null;
+    }
+
+    StorageDir selectedDir = dirCandidate.getFirst();
+    List<Long> victimBlocks = dirCandidate.getSecond();
+    Collections.reverse(victimBlocks);
+    if (selectedDir.getAvailableBytes() >= availableBytes) {
+      return new EvictionPlan(toMove, toEvict);
+    }
+
+    Map<StorageDir, Long> pendingBytesInDir = new HashMap<StorageDir, Long>();
+    for (long blockId : victimBlocks) {
+      StorageTier fromTier = mMeta.getBlockMeta(blockId).getParentDir().getParentTier();
+      List<StorageTier> toTiers = mMeta.getTiersBelow(fromTier.getTierAlias());
+      StorageDir toDir =
+          selectDirToMoveBlock(mMeta.getBlockMeta(blockId), toTiers, pendingBytesInDir);
+      if (toDir == null) {
+        // Not possible to Move
+        toEvict.add(blockId);
+      } else {
+        StorageTier toTier = toDir.getParentTier();
+        toMove.add(new Pair<Long, BlockStoreLocation>(blockId, new BlockStoreLocation(toTier
+            .getTierAlias(), toTier.getTierLevel(), toDir.getDirIndex())));
+        if (pendingBytesInDir.containsKey(toDir)) {
+          pendingBytesInDir.put(toDir, pendingBytesInDir.get(toDir)
+              + mMeta.getBlockMeta(blockId).getBlockSize());
+        } else {
+          pendingBytesInDir.put(toDir, mMeta.getBlockMeta(blockId).getBlockSize());
+        }
+      }
+    }
+    return new EvictionPlan(toMove, toEvict);
+  }
+
+  public Pair<StorageDir, List<Long>> getDirCandidate(long availableBytes,
```

Comment:
(1) Why this method is `public`? Is it supposed to be used by external classes? If not, make it `private`
(2)  Could you also javadoc this method?